### PR TITLE
Switching to pre-commit package. Fixes #1006

### LINKED ISSUE #1006: Switch to using pre-commit package for running linter and doc builder
**Description**

 

As a DCP member who makes metadata schemas, I would like to use the
 [`pre-commit`](https://pre-commit.com/) package rather than relying on the git hooks in order to automate running the schema linter. An example could be found [here](https://github.com/HumanCellAtlas/pipeline-tools/blob/master/.pre-commit-config.yaml).

Thanks @rexwangcc !

**Acceptance Criteria**

  
- [x] schema linter running using pre-commit package
- [x] doc builder running using pre-commit package

### PR DESCRIPTION
**Pull request by ESapenaVentura**
### Release notes

**Files created:**
- `.pre-commit-config.yaml`
- `.pre-commit-hooks.yaml`
- `src/add_docs.sh`

**Files modified:**
- `src/schema_linter.py`: Path used inside script now takes in account if you're in the root or in the src directory. 
- `src/human_readable_json.py`: Path used inside script now takes in account if you're in the root or in the src directory. 

**Files/folders that we plan on deleting:**
- `.githooks/pre-commit`
-`.githooks/`

**How does it work/How to review this feature:**
- Install `pre-commit` ([Webpage](https://pre-commit.com/), with pip)
- Now, whenever you commit, calls the hooks in the following order:
   - ~~`schema-linter`~~ The linter is no longer called.
   - `human_readable_json`: Creates the `md` documents.
   - `add_docs`: Simple executable that adds the created `md` documents to git for the commit. 
- You can see the process printed in the terminal. It will only output stdout for the scripts in case any of them fails.
- If everything went correctly, it will go forward with the commit.

In case you're reviewing this, pull the branch to your local directory, change something so the linter will fail (e.g., `biomaterial_id` property to `biomaterial_iddddd`), add and try to commit. It should fail.  

### Reviews requested

- Need 1 or 2 Reviewers to test that this works properly.
- Last Reviewer to review/approve, merge changes by 12/06/2019

### Extra acceptance criteria:
- People that wants to commit to this repo installs pre-commit.
- Issue to change documentation on repo is created.

### CODE REVIEW

**Comment by ESapenaVentura** [comment]
Don't try to test it for now. It's not working again.

**Comment by ESapenaVentura** [comment]
@HumanCellAtlas/wranglers @daniwelter I have assigned mallory for review, but any suggestion/help is appreciated!

**Comment by malloryfreeberg** [comment]
@ESapenaVentura will removing the `.githooks/` folder affect our hooks into Slack at all?

**Comment by ESapenaVentura** [comment]
> @ESapenaVentura will removing the `.githooks/` folder affect our hooks into Slack at all?

I have no idea how slack pull requests interact with slack, so I can't answer you that. I'll look into it and tell you what I can find!

**Comment by ESapenaVentura** [comment]
> @ESapenaVentura will removing the `.githooks/` folder affect our hooks into Slack at all?

`https://github.com/integrations/slack/blob/master/README.md`
`https://api.slack.com/incoming-webhooks`

It doesn't seem like uses githooks at all. In both cases, it's a separate app that checks on the repo! (Or, at least, that's what I understood).

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+fail_fast: true
+
+repos:
+- repo: local
+  hooks:
+    - id: schema_linter
```
Comment (daniwelter) [inline .pre-commit-config.yaml]: As per discussion in hca-metadata-internal Slack channel - remove schema linter from pre-commit, so it only runs in the travis build

Comment (ESapenaVentura) [reply .pre-commit-config.yaml]: Deleted! :)

**Comment by malloryfreeberg** [comment]
Is this PR done? Can we close the branch and the ticket? @ESapenaVentura

**Comment by ESapenaVentura** [comment]
Actually, I just discovered a problem with the pre-commit hooks as they are, so I could use this branch to make the changes. [Ticket](https://github.com/HumanCellAtlas/metadata-schema/issues/1025)

**Pull request merged**